js/imu.js

```diff
@@ -104,61 +104,62 @@ function IMU(copyFrom) {
                 Y: accSmooth[1],
                 Z: accSmooth[2]
             };
-    
+
         rotateVector(result, rpy);
-    
+
         result.Z -= acc_1G;
-    
+
         return result;
     }
-    
-    // Use the craft's estimated roll/pitch to compensate for the roll/pitch of the magnetometer reading 
+
+    // Use the craft's estimated roll/pitch to compensate for the roll/pitch of the magnetometer reading
     function calculateHeading(vec, roll, pitch) {
-        var 
+        var
             cosineRoll = Math.cos(roll),
             sineRoll = Math.sin(roll),
             cosinePitch = Math.cos(pitch),
             sinePitch = Math.sin(pitch),
-        
+
             headingX = vec.X * cosinePitch + vec.Y * sineRoll * sinePitch + vec.Z * sinePitch * cosineRoll,
             headingY = vec.Y * cosineRoll - vec.Z * sineRoll,
-            heading = Math.atan2(headingY, headingX) + magneticDeclination / 10.0 * RAD;
-    
-        if (heading < 0)
-            heading += 2 * Math.PI;
-    
+            heading = Math.atan2(headingY, headingX) + magneticDeclination / 10.0 * RAD; // RAD = pi/180
+
+            heading += 2 * Math.PI; // positive all the time, we want zero to return pi
+            if (heading > 2* Math.PI)
+               heading -= 2* Math.PI;
+
```

Comment:
```suggestion
            if (heading > 2 * Math.PI) {
               heading -= 2 * Math.PI;
            }

```

Reply:
thanks, wasn't sure of the need for the braces